governance-aepp/src/utils/wallet.js

```diff
@@ -0,0 +1,131 @@
+}, {
+  name: 'ae_uat', instance: new Node('https://testnet.aeternity.io')
+}]
+
+export let sdk = null
+
+export const wallet = reactive({
+  activeWallet: null, address: null, balance: null, walletStatus: null, isStatic: false, networkId: null,
+})
+
+export const initWallet = async (eventBus) => {
+  const {walletStatus, isStatic, networkId} = toRefs(wallet)
+
+  walletStatus.value = 'connecting'
+
+  try {
+    // try to connect to Superhero Wallet
+    sdk = new AeSdkAepp({
+      name: 'AEPP', nodes, compilerUrl: settings.compilerUrl, onNetworkChange: async (network) => {
+        console.info('onNetworkChange:', network)
+        await aeConnectToNode(network.networkId)
+        if (eventBus) eventBus.emit('dataChange');
+      }, onAddressChange: async (addresses) => {
+        console.info('onAddressChange:', addresses)
+        await aeConnectToNode(networkId.value)
+        if (eventBus) eventBus.emit('dataChange');
+      },
+    })
+
+    await scanForWallets()
+
+    if (walletStatus.value === 'fallback_static') {
+      isStatic.value = true;
+
+      sdk = new AeSdk({
+        compilerUrl: settings.compilerUrl, nodes,
+      })
+
+      await aeConnectToNode(settings.defaultNetworkId)
+    }
+  } catch (error) {
+    console.error('initWallet error:', error)
+    return false
+  }
+
+  return true
+}
+
+export const scanForWallets = async () => {
+  const {walletStatus, activeWallet} = toRefs(wallet)
+
+  walletStatus.value = 'scanning'
+
+  const detectedWallet = await Promise.race([new Promise((resolve) => {
+    const handleWallets = async () => {
+      stopScan()
+      resolve(true);
+    }
+
+    const scannerConnection = new BrowserWindowMessageConnection()
+    const stopScan = walletDetector(scannerConnection, handleWallets)
+  }), new Promise((resolve) => setTimeout(() => {
+    resolve(false);
+  }, 1000))]);
+
+  console.log('scanForWallets detectedWallet:', detectedWallet)
+
+  if (detectedWallet) {
+    await new Promise((resolve) => {
+      const handleWallets = async ({wallets, newWallet}) => {
+        try {
+          walletStatus.value = 'asking_permission'
+          newWallet = newWallet || Object.values(wallets)[0]
+          stopScan()
+
+          activeWallet.value = newWallet
+
+          const {networkId} = await sdk.connectToWallet(newWallet.getConnection())
+          await sdk.subscribeAddress('subscribe', 'current')
+
+          await aeConnectToNode(networkId);
+          resolve()
+        } catch (error) {
+          console.error('scanForWallets error:', error)
+          walletStatus.value = 'fallback_static'
+        }
+      }
+
+      const scannerConnection = new BrowserWindowMessageConnection()
+      const stopScan = walletDetector(scannerConnection, handleWallets)
+    })
+  } else {
+    walletStatus.value = 'fallback_static'
+  }
+}
+
+export const fetchWalletInfo = async () => {
+  const {address, balance, walletStatus, isStatic, networkId} = toRefs(wallet)
```

Comment:
if we access the wallet object directly it should not be needed

Reply:
is it still reactive then? this is the pattern I followed from the boilerplate

Reply:
https://github.com/aeternity/aepp-boilerplate-vue/blob/master/src/utils/aeternity/wallet.js